source/crud/tests/unified/estimatedDocumentCount.yml

```diff
@@ -0,0 +1,276 @@
+    operations:
+      - name: estimatedDocumentCount
+        object: *collection0
+        arguments:
+          maxTimeMS: 6000
+        expectResult: 3
+    expectEvents:
+      - client: *client0
+        events:
+          - commandStartedEvent:
+              command:
+                aggregate: *collection0Name
+                pipeline: &pipeline
+                  - $collStats: { count: {} }
+                  - $group: {
+                    _id: 1,
+                    n: { $sum: $count },
+                  }
+                maxTimeMS: 6000
+              commandName: aggregate
+              databaseName: *database0Name
+
+  - description: "estimatedDocumentCount on non-existent collection on 4.9.0 or greater"
+    runOnRequirements:
+      - minServerVersion: "4.9.0"
+    operations:
+      - name: estimatedDocumentCount
+        object: *collection1
+        expectResult: 0
+    expectEvents:
+      - client: *client0
+        events:
+          - commandStartedEvent:
+              command:
+                aggregate: *collection1Name
+                pipeline: &pipeline
+                  - $collStats: { count: {} }
+                  - $group: {
+                    _id: 1,
+                    n: { $sum: $count },
+                  }
+              commandName: aggregate
+              databaseName: *database0Name
+
+  - description: "estimatedDocumentCount errors correctly on 4.9.0 or greater--command error"
+    runOnRequirements:
+      - minServerVersion: "4.9.0"
+    operations:
+      - name: failPoint
+        object: testRunner
+        arguments:
+          client: *client0
+          failPoint:
+            configureFailPoint: failCommand
+            mode: { times: 1 }
+            data:
+              failCommands: [ aggregate ]
+              errorCode: 11000
+      - name: estimatedDocumentCount
+        object: *collection0
+        expectError: { isError: true }
+    expectEvents:
+      - client: *client0
+        events:
+          - commandStartedEvent:
+              command:
+                aggregate: *collection0Name
+                pipeline: &pipeline
+                  - $collStats: { count: {} }
+                  - $group: {
+                    _id: 1,
+                    n: { $sum: $count },
+                  }
+              commandName: aggregate
+              databaseName: *database0Name
+
+  - description: "estimatedDocumentCount errors correctly on 4.9.0 or greater--socket error"
+    runOnRequirements:
+      - minServerVersion: "4.9.0"
+    operations:
+      - name: failPoint
+        object: testRunner
+        arguments:
+          client: *client0
+          failPoint:
+            configureFailPoint: failCommand
+            mode: { times: 1 }
+            data:
+              failCommands: [ aggregate ]
+              closeConnection: true
+      - name: estimatedDocumentCount
+        object: *collection0
+        expectError: { isError: true }
```

Comment:
Since we're expecting a network error, I propose using a strict assertion:

```yaml
expectError:
  isClientError: true
```

Existing unified spec tests only used this assertion for argument validation errors, but [expectError](https://github.com/mongodb/specifications/blob/master/source/unified-test-format/unified-test-format.rst#expectederror) states:

> Client errors include, but are not limited to: parameter validation errors before a command is sent to the server; network errors.

This would be a good opportunity to test that unified test runner implementations actually consider network errors to be client errors (i.e. it is not derived from a server response).

Reply:
The `isClientError: true` assertion seems to fail on the Go test runner with
```
expected error connection(localhost:27017[-21]) socket was unexpectedly closed: EOF to be a client error: true, is client error: false
```
I'm not sure if this kind of network error (unexpected socket closure) qualifies as a client error.

Keeping just `isError` for now.

Reply:
I created [DRIVERS-1555](https://jira.mongodb.org/browse/DRIVERS-1555) to track this, and left a todo to come back and update this test at a later point.